iothub/device/src/ModuleClient.cs

```diff
@@ -359,6 +373,88 @@ public void SetRetryPolicy(IRetryPolicy retryPolicy)
             this.internalClient.SetMessageHandlerAsync(messageHandler, userContext);
 
+        /// <summary>
+        /// Interactively invokes a method on device
+        /// </summary>
+        /// <param name="deviceId">Device Id</param>
+        /// <param name="directMethodRequest">Device method parameters (passthrough to device)</param>
+        /// <returns>Method result</returns>
+        public Task<DirectMethodResult> InvokeMethodAsync(string deviceId, DirectMethodRequest directMethodRequest)
+        {
+            return this.InvokeMethodAsync(deviceId, directMethodRequest, CancellationToken.None);
+        }
+
+        /// <summary>
+        /// Interactively invokes a method on device
+        /// </summary>
+        /// <param name="deviceId">Device Id</param>
+        /// <param name="directMethodRequest">Device method parameters (passthrough to device)</param>
+        /// <param name="cancellationToken">Cancellation Token</param>
+        /// <returns>Method result</returns>
+        public Task<DirectMethodResult> InvokeMethodAsync(string deviceId, DirectMethodRequest directMethodRequest, CancellationToken cancellationToken)
+        {
+            return InvokeMethodAsync(GetDeviceMethodUri(deviceId), directMethodRequest, cancellationToken);
+        }
+
+        /// <summary>
+        /// Interactively invokes a method on module
+        /// </summary>
+        /// <param name="deviceId">Device Id</param>
+        /// <param name="moduleId">Module Id</param>
+        /// <param name="directMethodRequest">Device method parameters (passthrough to device)</param>
+        /// <returns>Method result</returns>
+        public Task<DirectMethodResult> InvokeMethodAsync(string deviceId, string moduleId, DirectMethodRequest directMethodRequest)
+        {
+            return this.InvokeMethodAsync(deviceId, moduleId, directMethodRequest, CancellationToken.None);
+        }
+
+        /// <summary>
+        /// Interactively invokes a method on module
+        /// </summary>
+        /// <param name="deviceId">Device Id</param>
+        /// <param name="moduleId">Module Id</param>
+        /// <param name="directMethodRequest">Device method parameters (passthrough to device)</param>
+        /// <param name="cancellationToken">Cancellation Token</param>
+        /// <returns>Method result</returns>
+        public Task<DirectMethodResult> InvokeMethodAsync(string deviceId, string moduleId, DirectMethodRequest directMethodRequest, CancellationToken cancellationToken)
+        {
+            return InvokeMethodAsync(GetModuleMethodUri(deviceId, moduleId), directMethodRequest, cancellationToken);
+        }
+
+        private Task<DirectMethodResult> InvokeMethodAsync(Uri uri, DirectMethodRequest directMethodRequest, CancellationToken cancellationToken)
+        {
+            var httpClientHandler = new HttpClientHandler();
+
+#if NETSTANDARD1_3 || NETSTANDARD2_0
```

Comment:
How does cert validation work in the other cases?

Reply:
I added WebRequestHandler for dotnet framework

---
In reply to: [193912087](https://github.com/Azure/azure-iot-sdk-csharp/pull/501#discussion_r193912087) [](ancestors = 193912087)